src/classic/element/ReactElement.js

```diff
@@ -103,7 +103,7 @@ var ReactElement = function(type, key, ref, owner, context, props) {
   // through the owner.
   this._context = context;
 
-  if (__DEV__) {
+  if (__DEBUG__) {
```

Comment:
React 0.13 is deprecating prop mutations, and killing it in later versions. I figured those would be really nice warnings to have, but yeah -- it's pretty costly. I want to remember that that check happens in `ReactElementValidator`, which is why i kept that stuff around too. But I'll take another round to make absolutely sure before including that.

Keeping prop validation errors is very likely to cause flooding (we have a lot of them), so this might be to costly to keep anyway in case prop mutations and prop validations comes as a package.